pages/api/refresh-msca.js

```diff
@@ -3,25 +3,41 @@
  *
  */
 
-import { AuthIsDisabled, AuthIsValid } from '../../lib/auth'
+import { AuthIsDisabled, AuthIsValid, ValidateSession } from '../../lib/auth'
 import { getLogger } from '../../logging/log-util'
 import { authOptions } from 'pages/api/auth/[...nextauth]'
 import { getServerSession } from 'next-auth/next'
+import { getToken } from 'next-auth/jwt'
+
+// Including crypto module
+const crypto = require('crypto')
 
 //The below sets the minimum logging level to error and surpresses everything below that
 const logger = getLogger('refresh-msca')
 logger.level = 'error'
 
 export default async function handler(req, res) {
   const session = await getServerSession(req, res, authOptions)
+  const token = await getToken({ req })
+  //Generate a random id for each request to ensure unique responses/no caching
+  const id = crypto.randomBytes(20).toString('hex')
 
   if (req.method === 'GET') {
     //Send request to ECAS to refresh MSCA session
     if (AuthIsDisabled()) {
       //Service unavailable when auth is disabled
       res.status(503).json({ success: false })
     } else if (await AuthIsValid(req, session)) {
-      res.status(200).json({ success: true })
+      //If auth session is valid, make GET request to validateSession endpoint
+      const sessionValid = await ValidateSession(
+        process.env.CLIENT_ID,
+        token.sub
+      )
+      if (sessionValid) {
+        res.status(200).json({ success: sessionValid, id: id })
+      } else {
+        res.status(302).json({ success: sessionValid, id: id })
```

Comment:
Am I reading this correctly: if the session is not valid, return a 302: found?

Reply:
Updated to `401 unauthorized` based on your comments which makes more sense. I had originally used 302 for a temporary redirect